dev-system/dev/user.clj

```diff
@@ -52,55 +75,64 @@
-  ;; so that ACLs that are bootstrapped in kernel reference same Administrators group in bootstrapped CMR.
+  ;; Hard coded here and here:
+  ;; * legacy-services/echo/echo-env/support/db/bootstrap/business/system_acls_changeLog.xml
+  ;; so that ACLs that are bootstrapped in kernel reference same Administrators group in
+  ;; bootstrapped CMR.
   (transmit-config/set-administrators-group-legacy-guid! "316520E041894014E050007F010038C4"))
 
 (defn start
-  "Starts the current development system."
-  []
-  (config/reset-config-values)
-
-  (jobs/set-default-job-start-delay! (* 3 3600))
-
-  (system/set-gorilla-repl-port! 8090)
-
-  ;; Comment/uncomment these lines to switch between external and internal settings.
-
-  (system/set-dev-system-elastic-type! :in-memory)
-  ; (system/set-dev-system-elastic-type! :external)
-
-  ;; Note external ECHO does not work with the automated tests. The automated tests expect they
-  ;; can interact with the Mock ECHO to setup users, acls, and other ECHO objects.
-  (system/set-dev-system-echo-type! :in-memory)
-  ; (system/set-dev-system-echo-type! :external)
-
-
-  ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
-  ;; MAKE SURE YOU DISABLE SYMANTEC ANTIVIRUS BEFORE STARTING THE TESTS WITH EXTERNAL DB
-  ;; Renable them when you're done
-
-  (system/set-dev-system-db-type! :in-memory)
-  ; (system/set-dev-system-db-type! :external)
-
-  ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
-
-  (system/set-dev-system-message-queue-type! :in-memory)
-  ; (system/set-dev-system-message-queue-type! :external)
-  ; (system/set-dev-system-message-queue-type! :aws)
-
-  (sit-sys/set-logging-level @settings/logging-level)
-
-  ;; Uncomment this to force CMR to use Legacy Services
-  ; (configure-for-legacy-services)
-
-  (let [s (system/create-system)
-        s (configure-systems-logging s @settings/logging-level)]
-    (alter-var-root #'system
-                    (constantly
-                      (system/start s))))
-
-  (d/touch-user-clj))
+  "Starts the current development system.
+
+  If a mode value is not passed, it will use whatever mode has already been
+  set. If no mode has been set, the ns-level `mode` variable will be set to
+  `:in-memory`.
+
+  Note that when an explicit mode is passed as an argument, not only will CMR
+  be started in that mode, but the ns-level `user/mode` variable will be reset
+  to the given value."
+  ([]
+    (start (if (or (nil? mode) (empty? mode))
+             :in-memory
+             mode)))
+  ([new-mode]
+    (set-mode! new-mode)
+    (config/reset-config-values)
+
+    (jobs/set-default-job-start-delay! (* 3 3600))
+
+    (system/set-gorilla-repl-port! 8090)
+
+    (case mode
+      :in-memory (info "Running CMR in-memory ...")
+      :external (info "Running CMR externally ..."))
+    (system/set-dev-system-elastic-type! mode)
```

Comment:
Hey Chris, wanted to live a response to this comment, in case you'd missed it: I've pushed updates which add support for setting individual run modes.